### ISSUE TITLE
v5.0.4 M1.4.7 MCPC-Plus-248 - getOnlinePlayers() NoSuchMethodError

### ISSUE THREAD

**Issue by iMasonite** [kangarko/ChatControl#73]
When looking through the source I noticed that in a few locations there is 
<code>Bukkit.getOnlinePlayers().size()</code>
instead of
<code>Bukkit.getOnlinePlayers().length</code>

with compile error
<code>Cannot invoke size() on the array type Player[]</code>

Log details:

    2015-02-02 10:10:31 [INFO] This server is running CraftBukkit version git-MCPC-Plus-jenkins-MCPC-Plus-248 
    (MC: 1.4.7) (Implementing API version 1.4.7-R1.1-SNAPSHOT)

    2015-02-02 10:10:42 [INFO] [ChatControl] Enabling ChatControl v5.0.4
    2015-02-02 10:10:42 [INFO] Created default file: settings.yml
    2015-02-02 10:10:42 [INFO] Created default file: rules.txt
    2015-02-02 10:10:42 [INFO] Created default file: handlers.yml
    2015-02-02 10:10:42 [INFO] Created default file: chat.txt
    2015-02-02 10:10:42 [INFO] Created default file: commands.txt
    2015-02-02 10:10:42 [INFO] Created default file: sign.txt
    2015-02-02 10:10:42 [INFO] Created default file: packets.txt
    2015-02-02 10:10:42 [INFO] Loaded 17 Rules in chat.txt
    2015-02-02 10:10:42 [INFO] Loaded 1 Rules in sign.txt
    2015-02-02 10:10:42 [INFO] Loaded 3 Rules in commands.txt
    2015-02-02 10:10:42 [INFO] Loaded 1 Rules in packets.txt
    2015-02-02 10:10:42 [INFO] Loaded 6 Rules in rules.txt
    2015-02-02 10:10:42 [SEVERE] java.lang.NoSuchMethodError: org.bukkit.Server.getOnlinePlayers()Ljava/util/Collection;
    2015-02-02 10:10:42 [SEVERE] 	at kangarko.chatcontrol.ChatControl.onEnable(Unknown Source)
    2015-02-02 10:10:42 [SEVERE] 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:217)
    2015-02-02 10:10:42 [SEVERE] 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:396)
    2015-02-02 10:10:42 [SEVERE] 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:381)
    2015-02-02 10:10:42 [SEVERE] 	at org.bukkit.craftbukkit.v1_4_R1.CraftServer.loadPlugin(CraftServer.java:291)
    2015-02-02 10:10:42 [SEVERE] 	at org.bukkit.craftbukkit.v1_4_R1.CraftServer.enablePlugins(CraftServer.java:273)
    2015-02-02 10:10:42 [SEVERE] 	at net.minecraft.server.MinecraftServer.j(MinecraftServer.java:602)
    2015-02-02 10:10:42 [SEVERE] 	at net.minecraft.server.MinecraftServer.e(MinecraftServer.java:568)
    2015-02-02 10:10:42 [SEVERE] 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:447)
    2015-02-02 10:10:42 [SEVERE] 	at ho.c(DedicatedServer.java:203)
    2015-02-02 10:10:42 [SEVERE] 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:172)
    2015-02-02 10:10:42 [SEVERE] 	at fy.run(ThreadMinecraftServer.java:16)
    2015-02-02 10:10:42 [INFO] !----------------------------------------------!
    2015-02-02 10:10:42 [INFO]  Error loading ChatControl, plugin is disabled!
    2015-02-02 10:10:42 [INFO]  Running on server v1_4_R1 and Java 1.7.0_71
    2015-02-02 10:10:42 [INFO] !----------------------------------------------!
    2015-02-02 10:10:42 [INFO]  The error was: org.bukkit.Server.getOnlinePlayers()Ljava/util/Collection;
    2015-02-02 10:10:42 [INFO] !----------------------------------------------!
    2015-02-02 10:10:42 [INFO] [ChatControl] Disabling ChatControl v5.0.4

**Comment by kangarko** [comment]
Are you using the compiled version or compiling by yourself?

The API has changed but there should be backward compatibility, at least it was in 5.0.2 when I tested it last on 1.4.7.

**Comment by iMasonite** [comment]
Compiled version first and then compiled it with some changes, the changes to the afore mentioned worked, I had to also change the hook to Essentials as the CommandSource wasn't introduced in the version you've used, the 1.4.7 recommended build of essentials that is. There is a fork I've played around with on my git, messed up a little but got there in the end. My code formatter didn't like the way to set up the config system and removed unused local fields, but I'm sure you'll see what I did there just to get it working on the version I'm using. 

Super props to you on this new system. It's damn good.
